### PR TITLE
Rc manage sheet

### PR DESCRIPTION
**Pull request by robert-czwartosz**
Hey, I've done grade sheet view.
After clicking on grade sheet in table, it shows manage sheet view.

### CODE REVIEW

**Comment by fszwedo** [comment]
I'm receiving error for 
https://coders-camp-organization-app.herokuapp.com/api/grade/sheets/6050deed5721510015702663/mentor/grades
HTTP 400, message "value" failed custom validation because some grade is invalid"

seems like some grade validation is missing

**Comment by fszwedo** [comment]
@robert-czwartosz it's happening only when I add the grace comment/description - with those fields not-filled everything seems to be fine

awesome job btw with getting this done

**Comment by fszwedo** [comment]
@robert-czwartosz ok it just happens for one random project I opened, for others it works just fine :) don't have a clean reproduction path now

**Comment by robert-czwartosz** [comment]
Now, I reproduced this bug.
When you change project ---, its 404 error in patch request

**Comment by fszwedo** [comment]
@robert-czwartosz i also see something funky when you add a mentor grade - sometimes the previous grade is being removed or one new empty is being added

File: frontend/src/components/EditGradeModal/EditGrade.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  const handleSaveButtonClick = async () => {
+    const grade = {
+      [!!quality ? quality : props.quality]: {
+        points,
+        description,
+        comment,
+      }
+    };
+
+    console.log(grade);
```
Comment (fszwedo) [inline frontend/src/components/EditGradeModal/EditGrade.tsx:102]: to be removed after troubleshooting is finished

File: frontend/src/components/ManageSheet/ManageSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+      .then(loadSheet)
+      .catch(error => console.log(error));
+  }
+
+  const handleParticipantSelection = (row: GridSelectionModelChangeParams) => {
+    selectedParticipants.current = row.selectionModel as string[];
+  }
+
+  const deleteSelectedUsers = () => {
+    console.log(selectedParticipants.current);
```
Comment (fszwedo) [inline frontend/src/components/ManageSheet/ManageSheet.tsx:144]: to be removed after troubleshooting

File: frontend/src/components/ManageSheet/ManageSheet.tsx [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+  const handleGradeSelection = (row: GridSelectionModelChangeParams) => {
+    selectedGrades.current = row.selectionModel as string[];
+  }
+
+  const handleEditGrade = (grade: Grades) => {
+    const newGrades = {
+      ...mentorGrades,
+      ...grade,
+    }
+    console.log(Object.keys(grade))
```
Comment (fszwedo) [inline frontend/src/components/ManageSheet/ManageSheet.tsx:159]: to be removed after troubleshooting

File: frontend/src/components/ManageSheet/ManageSheet.tsx [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+    if (editedGrade.quality !== Object.keys(grade)[0])
+      delete newGrades[editedGrade.quality];
+    api.patchMentorGrade(sheetId, newGrades)
+      .then( () => dispatch(fetchData(mentorGradesTableName, getMentorGrades)) )
+      .catch(error => console.log(error));
+  }
+
+  const deleteSelectedGrades = () => {
+    const newMentorGrades = _.omit(mentorGrades, selectedGrades.current)
+    console.log(newMentorGrades);
```
Comment (fszwedo) [inline frontend/src/components/ManageSheet/ManageSheet.tsx:169]: to be removed after troubleshooting

**Review by fszwedo — APPROVED** [review]
apart from the thing with the grades I don't see any issues, everything works good 👍 
if this will be not debuggable then feel free to merge this to main

**Pull request merged**